libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/ManeuverUtils.java

```diff
@@ -7,22 +7,29 @@
 import com.mapbox.services.android.navigation.R;
 
 import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_LEFT;
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_MODIFIER_UTURN;
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_ROTARY;
+import static com.mapbox.services.android.navigation.v5.navigation.NavigationConstants.STEP_MANEUVER_TYPE_ROUNDABOUT;
 
 public class ManeuverUtils {
-  private static final String EMPTY = "";
 
   public static int getManeuverResource(LegStep step) {
     ManeuverMap maneuverMap = new ManeuverMap();
     if (step != null && step.maneuver() != null) {
       StepManeuver maneuver = step.maneuver();
-      if (!TextUtils.isEmpty(maneuver.modifier())) {
+      String maneuverType = maneuver.type();
+      String maneuverModifier = maneuver.modifier();
+      if (!TextUtils.isEmpty(maneuverModifier)) {
         String drivingSide = step.drivingSide();
-        if (!STEP_MANEUVER_MODIFIER_LEFT.equals(drivingSide)) {
-          drivingSide = EMPTY;
+        if (STEP_MANEUVER_MODIFIER_LEFT.equals(drivingSide)) {
+          if (STEP_MANEUVER_MODIFIER_UTURN.equals(maneuverModifier)
```

Comment:
i wonder if we could reduce number (5) of nested `if` conditions to improve readability of this method.